src/modules/netinstall/PackageModel.cpp

```diff
@@ -309,3 +341,27 @@ PackageModel::setupModelData( const QVariantList& l )
     setupModelData( l, m_rootItem );
     emit endResetModel();
 }
+
+void
+PackageModel::appendModelData( const QVariantList& groupList, const QString source )
+{
+    if ( m_rootItem )
+    {
+        emit beginResetModel();
+
+        // Prune any existing data from the same source
+        for ( int i = 0; i < m_rootItem->childCount(); i++ )
+        {
+            PackageTreeItem* child = m_rootItem->child( i );
+            if ( child->source() == source )
+            {
+                m_rootItem->removeChild( i );
+            }
+        }
+
+        // Add the new data to the model
+        setupModelData( groupList, m_rootItem );
+
+        emit endResetModel();
```

Comment:
Please use `Q_EMIT` (I think it's ugly, but it is the way forward to "unclaim" `emit` as a special-for-Qt keyword)